src/controllers/revision.ts

```diff
@@ -570,13 +575,17 @@ export const downloadRevisionCubeAsJSON = async (req: Request, res: Response, ne
   const { dataset, revision } = res.locals;
   const lang = req.language.split('-')[0];
   let cubeFile: string;
-  if (revision.onlineCubeFilename) {
+  if (revision.onlineCubeFilename && !revision.onlineCubeFilename.includes('proto')) {
     const fileBuffer = await req.fileService.loadBuffer(revision.onlineCubeFilename, dataset.id);
     cubeFile = tmp.tmpNameSync({ postfix: '.duckdb' });
     fs.writeFileSync(cubeFile, fileBuffer);
+  } else if (revision.onlineCubeFilename && revision.onlineCubeFilename.includes('proto')) {
+    const fileBuffer = await req.fileService.loadBuffer(revision.onlineCubeFilename, dataset.id);
+    cubeFile = tmp.tmpNameSync({ postfix: '.duckdb' });
+    fs.writeFileSync(cubeFile, fileBuffer);
+    cubeFile = await createBaseCubeFromProtoCube(dataset.id, revision, cubeFile);
```

Comment:
There's a lot of duplicated code in these download handlers already that feels ripe for refactoring, but let's try and avoid adding more :D 

```suggestion
  if (revision.onlineCubeFilename) {
    const fileBuffer = await req.fileService.loadBuffer(revision.onlineCubeFilename, dataset.id);
    cubeFile = tmp.tmpNameSync({ postfix: '.duckdb' });
    fs.writeFileSync(cubeFile, fileBuffer);
    if (revision.onlineCubeFilename.includes('proto')) {
      cubeFile = await createBaseCubeFromProtoCube(dataset.id, revision, cubeFile);
    }
```

same pattern should work for the other 4 below I think?

Reply:
Yeah... I think I need to refactor the code out once I have sometime to do so.